geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractLTrimIntegrationTest.java

```diff
@@ -0,0 +1,228 @@
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, 2);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(3);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 1, -2);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e2");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, -2, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(2);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, -1, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(1);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e1");
+  }
+
+  @Test
+  public void trimsToCorrectRange_givenSpecifiersOutsideListSize() {
+    initializeTestList();
+
+    jedis.ltrim(KEY, -4, -1);
+    assertThat(jedis.llen(KEY)).isEqualTo(4);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 3)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, -10, 10);
+    assertThat(jedis.llen(KEY)).isEqualTo(4);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 3)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 0, 4);
+    assertThat(jedis.llen(KEY)).isEqualTo(4);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 3)).isEqualTo("e1");
+
+    initializeTestList();
+
+    jedis.ltrim(KEY, 0, 10);
+    assertThat(jedis.llen(KEY)).isEqualTo(4);
+    assertThat(jedis.lindex(KEY, 0)).isEqualTo("e4");
+    assertThat(jedis.lindex(KEY, 1)).isEqualTo("e3");
+    assertThat(jedis.lindex(KEY, 2)).isEqualTo("e2");
+    assertThat(jedis.lindex(KEY, 3)).isEqualTo("e1");
+  }
+
+  private void initializeTestList() {
+    jedis.del(KEY);
+    jedis.lpush(KEY, "e1", "e2", "e3", "e4");
+  }
+
+  @Test
+  public void removesKey_whenLastElementRemoved() {
+    final String keyWithTagForKeysCommand = "{tag}" + KEY;
+    jedis.lpush(keyWithTagForKeysCommand, "e1", "e2", "e3");
+
+    jedis.ltrim(keyWithTagForKeysCommand, 0, -4);
+    assertThat(jedis.llen(keyWithTagForKeysCommand)).isEqualTo(0L);
+    assertThat(jedis.keys(keyWithTagForKeysCommand)).isEmpty();
```

Comment:
I think you can just call `assertThat(jedis.exists(keyWithTagForKeysCommand)).isTrue()`. Doesn't make a big difference but I think it's nicer for readability. The next test has this too.

Reply:
Done for both